bundles/org.openhab.binding.etherrain/src/main/java/org/openhab/binding/etherrain/internal/api/EtherRainCommunication.java

```diff
@@ -0,0 +1,249 @@
+                        break;
+                    case "rz":
+                        response.setLastCommandResult(EtherRainCommandResult.fromString(status));
+                        break;
+                    case "ri":
+                        response.setLastActiveValue(Integer.parseInt(status));
+                        break;
+                    case "rn":
+                        response.setRainSensor(Integer.parseInt(status) == 1 ? true : false);
+                        break;
+                    default:
+                        logger.debug("Unknown response: {}", command);
+                }
+            }
+        }
+
+        return response;
+    }
+
+    public boolean commandIrrigate(int delay, int zone1, int zone2, int zone3, int zone4, int zone5, int zone6,
+            int zone7, int zone8) {
+        try {
+            if (sendGet("result.cgi?xi=" + delay + ":" + zone1 + ":" + zone2 + ":" + zone3 + ":" + zone4 + ":" + zone5
+                    + ":" + zone6 + ":" + zone7 + ":" + zone8) == null) {
+                return false;
+            }
+        } catch (IOException e) {
+            logger.warn("Could not send irrigate command to etherrain: {}", e.getMessage());
+            return false;
+        }
+
+        return true;
+    }
+
+    public boolean commandClear() {
+        try {
+            if (sendGet("/result.cgi?xr") == null) {
+                return false;
+            }
+        } catch (IOException e) {
+            logger.warn("Could not send clear command to etherrain: {}", e.getMessage());
+            return false;
+        }
+
+        return true;
+    }
+
+    public boolean commandLogin() throws IOException {
+        return sendGet("/ergetcfg.cgi?lu=" + ETHERRAIN_USERNAME + "&lp=" + password) != null;
+    }
+
+    public boolean commandLogout() {
+        try {
+            if (sendGet("/ergetcfg.cgi?m=o") == null) {
+                return false;
+            }
+        } catch (IOException e) {
+            logger.warn("Could not send logout command to etherrain: {}", e.getMessage());
+            return false;
+        }
+
+        return true;
+    }
+
+    private List<String> sendGet(String command) throws IOException {
+        String url = "http://" + address + ":" + port + "/" + command;
+
+        ContentResponse response;
+
+        try {
+            response = httpClient.newRequest(url).timeout(HTTP_TIMEOUT, TimeUnit.SECONDS).send();
+            if (response.getStatus() != HttpURLConnection.HTTP_OK) {
+                logger.warn("Etherrain return status other than HTTP_OK : {}", response.getStatus());
+                return null;
+            }
+        } catch (InterruptedException | TimeoutException | ExecutionException e) {
+            logger.warn("Could not connect to Etherrain with exception: {}", e.getMessage());
+            return null;
+        }
+
+        List<String> rVal = new BufferedReader(new StringReader(response.getContentAsString())).lines()
```

Comment:
This feels very indirect converting the `String `to a `BufferedReader `to convert it to lines, what about using `Arrays.asList(string.split("\\s+"))`

Reply:
I tried your way, but it doesn't seem to work. The response is sent as an html file which gets dumped into the string and the file is not very easy to parse.